homeassistant/components/broadlink/light.py

```diff
@@ -141,3 +144,103 @@ async def _async_set_state(self, state):
 
         self._update_state(state)
         self.async_write_ha_state()
+
+
+class BroadlinkLight2(BroadlinkEntity, LightEntity):
+    """Representation of a Broadlink light."""
+
+    def __init__(self, device):
+        """Initialize the light."""
+        super().__init__(device)
+        self._attr_name = f"{device.name} Light"
+        self._attr_unique_id = device.unique_id
+        self._attr_supported_color_modes = set()
+
+        data = self._coordinator.data
+
+        if {"hue", "saturation"}.issubset(data):
+            self._attr_supported_color_modes.add(COLOR_MODE_HS)
+        if "colortemp" in data:
+            self._attr_supported_color_modes.add(COLOR_MODE_COLOR_TEMP)
+        if not self.supported_color_modes:
+            self._attr_supported_color_modes = {COLOR_MODE_BRIGHTNESS}
+
+        self._update_state(data)
+
+    def _update_state(self, data):
+        """Update the state of the entity."""
+        if "pwr" in data:
+            self._attr_is_on = bool(data["pwr"])
+
+        if "brightness" in data:
+            self._attr_brightness = round(data["brightness"] * 2.55)
+
+        if self.supported_color_modes == {COLOR_MODE_BRIGHTNESS}:
+            self._attr_color_mode = COLOR_MODE_BRIGHTNESS
+            return
+
+        if {"hue", "saturation"}.issubset(data):
+            self._attr_hs_color = [data["hue"], data["saturation"]]
+
+        if "colortemp" in data:
+            self._attr_color_temp = round(500 - (data["colortemp"] - 2700) / 10.951)
+
+        if "bulb_colormode" in data:
+            if data["bulb_colormode"] == BROADLINK_COLOR_MODE_RGB:
+                self._attr_color_mode = COLOR_MODE_HS
+            elif data["bulb_colormode"] == BROADLINK_COLOR_MODE_WHITE:
+                self._attr_color_mode = COLOR_MODE_COLOR_TEMP
+            else:
+                # Scenes are not yet supported.
+                self._attr_color_mode = COLOR_MODE_UNKNOWN
+
+    async def async_turn_on(self, **kwargs):
+        """Turn on the light."""
+        state = {"pwr": 1}
+
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = kwargs[ATTR_BRIGHTNESS]
+            state["brightness"] = round(brightness / 2.55)
+
+        if self.supported_color_modes == {COLOR_MODE_BRIGHTNESS}:
+            state["bulb_colormode"] = BROADLINK_COLOR_MODE_WHITE
+
+        elif ATTR_HS_COLOR in kwargs:
+            hs_color = kwargs[ATTR_HS_COLOR]
+            state["hue"] = int(hs_color[0])
+            state["saturation"] = int(hs_color[1])
+            state["bulb_colormode"] = BROADLINK_COLOR_MODE_RGB
+
+        elif ATTR_COLOR_TEMP in kwargs:
+            color_temp = kwargs[ATTR_COLOR_TEMP]
+            state["colortemp"] = (500 - color_temp) * 10.951 + 2700
```

Comment:
Why these numbers? Please name them.

Reply:
HA's Color temperature slider starts from 153 till 500, with 153 being cool white and 500 being warm, but 2700 is warm and 6500 is cool. color_temp is value HA will pass to the integration, I wanted 500 to be equal to 2700 and 153 to be equal to 6500, so `((500-500)*10.951)+2700 = 2700` and `((500-153)*10.951)+2700 = 6499.997`. It was built on the logic from @L-I-Am but with correct values

Reply:
Awesome! We could name them `max_color_temp_hass`, `min_color_temp_hass`, `min_color_temp_kelvin`, `max_color_temp_kelvin` and make the calculation of 10.951 explicit.

Reply:
This is great, thank you for your contribution! 
Could you also change the temperature values to the correct values for the LB1 class while you’re at it?